### PR TITLE
DFS/BFS/Dijkstra and Travis.

### PR DESCRIPTION
**Pull request by AVParamonov**
DFS/BFS/Dijkstra search algorithms implementations added.
Travis config added.

### CODE REVIEW

**Review by zfrgt — CHANGES_REQUESTED** [review]
Please see comments

File: algorithms/src/main/java/com/aparamonov/algostructure/search/BFS.java [hunk 1]
```diff
@@ -14,8 +14,8 @@
 
     public List<Node<T>> search(Node<T> node) {
 
-        List<Node<T>> visited = new LinkedList<>();
-        List<Node<T>> unvisited = new LinkedList<>();
+        List<Node<T>> visited = new ArrayList<>();
```
Comment (zfrgt) [inline algorithms/src/main/java/com/aparamonov/algostructure/search/BFS.java]: i'm not really sure this change is necessary. you're working with *graph* structures, not arrays. please check the performance for linked and array-lists

**Review by zfrgt — CHANGES_REQUESTED** [review]
comments

File: algorithms/src/main/java/com/aparamonov/algostructure/search/BFS.java (hunk 1 above)
Comment (AVParamonov) [reply algorithms/src/main/java/com/aparamonov/algostructure/search/BFS.java]: Measured performance difference between those two variants and it seems that it is not significant if exists at all cause we use them in streams.

**Comment by AVParamonov** [comment]
@alexey-kotlyar please review.

**Comment by zfrgt** [comment]
@AVParamonov gj, will merge it now. one more point for practice - plz add the tests' results to the readme.md of the repository in table format

**Pull request merged**